tools/provision/debian.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# g++ stuff in wheezy has a bug which has been fixed in gcc 4.7.3 which conveniently is exactly 
+# the version that is not shipped in wheezy
+# See https://gcc.gnu.org/viewcvs/gcc/branches/gcc-4_7-branch/libstdc%2B%2B-v3/include/std/condition_variable?view=patch&r1=189276&r2=193528&pathrev=193528
+
+function patch_condition_variable()
+{
+	TARGET_DIR=/usr/include/c++/4.7
+	pushd $TARGET_DIR
+	
+	if [[ ! -f condition_variable.orig ]]; then		
+		sudo cp condition_variable condition_variable.orig
+		sudo patch -p5 < $FILES_DIR/debian/condition_variable.patch
```

Comment:
It's also possible for us to set an overriding `sysroot` by creating a new directory within:

https://github.com/facebook/osquery/tree/master/sysroots/

for wheezy, then mimicking the expected structure as if that `DIR=./sysroots/debian` was set as `-I${DIR}`. We do that for *all* Linux distributions already to get past a C++ compiler error with iptables libraries: https://github.com/facebook/osquery/blob/master/CMakeLists.txt#L331

For the specific version of Debian you can do a CMake check:

```cmake
if(${OSQUERY_BUILD_DISTRO} STREQUAL "wheezy")
  include_directories("${CMAKE_SOURCE_DIR}/sysroots/wheezy")
endif()